staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/fields_proto.go

```diff
@@ -27,13 +27,20 @@ import (
 // that matches Fields. Do not use in Go structs.
 type ProtoFields struct {
 	// Map is the representation used in the alpha version of this API
-	Map map[string]Fields `json:"-" protobuf:"bytes,1,rep,name=map"`
+	Map map[string]ProtoFields `json:"-" protobuf:"bytes,1,rep,name=map"`
 
 	// Raw is the underlying serialization of this object.
 	Raw []byte `json:"-" protobuf:"bytes,2,opt,name=raw"`
 }
 
+// MarshalJSON implements json.Marshaler
```

Comment:
this is a bit misleading/fragile... it only works on legacy objects where Map is set and Raw is not

Reply:
Hmm true. I can't think of a case where Map and Raw would both be set, but we should probably handle that somehow

Reply:
or give an error